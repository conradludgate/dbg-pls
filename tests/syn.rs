macro_rules! assert_pretty_syn_snapshot {
    ($ty:ty => $code:literal) => {
        let parsed = &syn::parse_str::<$ty>($code).unwrap();
        let format = format!(
            "pretty:\n{}\n\ncolor:\n{}\n",
            dbg_pls::pretty(&parsed),
            dbg_pls::color(&parsed)
        );
        insta::assert_snapshot!(insta::_macro_support::AutoName, &format, $code);
    };
}

#[test]
fn expr_array() {
    assert_pretty_syn_snapshot!(syn::Expr => r#"
            [
                "Hello, World! I am a long string",
                420,
                "Wait, you can't mix and match types in arrays, is this python?",
                69,
                "Nice."
            ]
        "#);
}

#[test]
fn qself() {
    assert_pretty_syn_snapshot!(syn::Expr => "<Foo as Bar::<<Baz as Qux>::Bang>>::Boo");
}

#[test]
fn item_macro() {
    assert_pretty_syn_snapshot!(syn::Item => r#"
            macro_rules! assert_pretty_syn_snapshot {
                ($ty:ty => $code:literal) => {
                    insta::assert_snapshot!(
                        insta::_macro_support::AutoName,
                        &crate::pretty(&syn::parse_str::<$ty>($code).unwrap()).to_string(),
                        $code
                    );
                };
            }
        "#);
}
