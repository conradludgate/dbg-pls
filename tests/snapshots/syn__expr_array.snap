---
source: tests/syn.rs
expression: "\n            [\n                \"Hello, World! I am a long string\",\n                420,\n                \"Wait, you can't mix and match types in arrays, is this python?\",\n                69,\n                \"Nice.\"\n            ]\n        "
---
pretty:
Array(ExprArray {
    attrs: [],
    bracket_token: Bracket,
    elems: [
        Lit(ExprLit {
            attrs: [],
            lit: Str(LitStr {
                token: "Hello, World! I am a long string",
            }),
        }),
        Comma,
        Lit(ExprLit {
            attrs: [],
            lit: Int(LitInt { token: 420 }),
        }),
        Comma,
        Lit(ExprLit {
            attrs: [],
            lit: Str(LitStr {
                token: "Wait, you can't mix and match types in arrays, is this python?",
            }),
        }),
        Comma,
        Lit(ExprLit {
            attrs: [],
            lit: Int(LitInt { token: 69 }),
        }),
        Comma,
        Lit(ExprLit {
            attrs: [],
            lit: Str(LitStr { token: "Nice." }),
        }),
    ],
})

color:

[32mArray[0m[31m([0m[32mExprArray[0m [31m{[0m
    attrs[31m:[0m [31m[[0m[31m][0m[31m,[0m
    bracket_token[31m:[0m [32mBracket[0m[31m,[0m
    elems[31m:[0m [31m[[0m
        [32mLit[0m[31m([0m[32mExprLit[0m [31m{[0m
            attrs[31m:[0m [31m[[0m[31m][0m[31m,[0m
            lit[31m:[0m [32mStr[0m[31m([0m[32mLitStr[0m [31m{[0m
                token[31m:[0m [33m[0m[31m"[0m[33mHello, World! I am a long string[0m[31m"[0m[33m[0m[31m,[0m
            [31m}[0m[31m)[0m[31m,[0m
        [31m}[0m[31m)[0m[31m,[0m
        [32mComma[0m[31m,[0m
        [32mLit[0m[31m([0m[32mExprLit[0m [31m{[0m
            attrs[31m:[0m [31m[[0m[31m][0m[31m,[0m
            lit[31m:[0m [32mInt[0m[31m([0m[32mLitInt[0m [31m{[0m token[31m:[0m 420 [31m}[0m[31m)[0m[31m,[0m
        [31m}[0m[31m)[0m[31m,[0m
        [32mComma[0m[31m,[0m
        [32mLit[0m[31m([0m[32mExprLit[0m [31m{[0m
            attrs[31m:[0m [31m[[0m[31m][0m[31m,[0m
            lit[31m:[0m [32mStr[0m[31m([0m[32mLitStr[0m [31m{[0m
                token[31m:[0m [33m[0m[31m"[0m[33mWait, you can't mix and match types in arrays, is this python?[0m[31m"[0m[33m[0m[31m,[0m
            [31m}[0m[31m)[0m[31m,[0m
        [31m}[0m[31m)[0m[31m,[0m
        [32mComma[0m[31m,[0m
        [32mLit[0m[31m([0m[32mExprLit[0m [31m{[0m
            attrs[31m:[0m [31m[[0m[31m][0m[31m,[0m
            lit[31m:[0m [32mInt[0m[31m([0m[32mLitInt[0m [31m{[0m token[31m:[0m 69 [31m}[0m[31m)[0m[31m,[0m
        [31m}[0m[31m)[0m[31m,[0m
        [32mComma[0m[31m,[0m
        [32mLit[0m[31m([0m[32mExprLit[0m [31m{[0m
            attrs[31m:[0m [31m[[0m[31m][0m[31m,[0m
            lit[31m:[0m [32mStr[0m[31m([0m[32mLitStr[0m [31m{[0m token[31m:[0m [33m[0m[31m"[0m[33mNice.[0m[31m"[0m[33m[0m [31m}[0m[31m)[0m[31m,[0m
        [31m}[0m[31m)[0m[31m,[0m
    [31m][0m[31m,[0m
[31m}[0m[31m)[0m

